### DISCUSSION TITLE
How can I contribute to the https://github.com/illuminate/queue repository?

### DISCUSSION THREAD

**Discussion by lvaneijden** [laravel/framework#49582] [Q&A] [answered]
Hey guys! I may have a silly question, but I found a bug in the QueueServiceProvider that resides in the https://github.com/illuminate/queue repository. 
When running a queue processing job, I would get the following error:
In QueueServiceProvider.php line 201:
                                                     
  Call to undefined method Monolog\Logger::driver()

I created an easy fix for it by checking if the driver function exists first, and would like to offer this up as a pull request / bugfix somewhere, but I can't figure out how to do it. Can anyone point me in the right direction?

**Comment by Jubeki** [comment, marked as answer]
You can contribute directly to https://github.com/laravel/framework because the code will later be split to other repositories.

The files you are looking for should be located here: https://github.com/laravel/framework/tree/10.x/src/Illuminate/Queue

**Reply by SocolaDaiCa** [reply]
amazing, how It split to other repositories?
I think they use .github/workflows to do that, but I cant see any workflow like that.

**Reply by Jubeki** [reply]
For Laravel it is not public how they exactly do it, but there are other repositories, were you can find out how to do it:

https://github.com/claudiodekker/laravel-auth/blob/master/.github/workflows/split.yml

https://github.com/filamentphp/filament/blob/3.x/.github/workflows/monorepo-split.yml

**Comment by lvaneijden** [comment]
Thanks for pointing out the location. I tried cloning the repository and creating a new branch based on the 10.x version to submit the patch, but it does not allow me to push my new branch. Do I need to name it specifically or follow a different route...?

**Reply by Jubeki** [reply]
You need to fork the project beforehand:
https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project

**Reply by lvaneijden** [reply]
Hmm I created a fork, followed the guide to create a branch and try to push from there, but it still does not allow me to push the branch (see https://github.com/lvaneijden/laravel-framework/branches/yours). I'm new to contributing to public repo's so there is a good chance I'm still doing it wrong. Pushing still tells me I'm not authorized or my settings are wrong (while I can still pull and push other repo's)

**Reply by Jubeki** [reply]
What is the result of `git remote get-url origin`, I think you are trying to push to the wrong repository, though I am not sure
?

**Reply by lvaneijden** [reply]
I get the following result: https://github.com/lvaneijden/laravel-framework.git

**Reply by Jubeki** [reply]
Can you send a screenshot of the error you are getting? Or the full error message?

**Reply by lvaneijden** [reply]
<img width="1302" alt="Screenshot 2024-01-10 at 09 32 18" src="https://github.com/laravel/framework/assets/9678159/ca5645ec-fc32-4435-bbd4-34c9f183b467">

It's this. I forked the repo by using git clone over https, maybe that is the issue?

**Reply by Jubeki** [reply]
That can be the issue, but should normally prompt you for your username and password.

As I only use the command line, I can not help with the GUI.
Maybe try the steps listed there. Or check how you have cloned other repositories.

As far as I am concerned, I don't think I will be able to help more.

**Reply by lvaneijden** [reply]
I was able to do it eventually, I needed to fork the repository, and then use ssh to clone that one locally. After then adding my ssh key, I was able to push and pull branches. Resulting in the following pull request: https://github.com/laravel/framework/pull/49635

**Comment by lvaneijden** [comment]
Pull request was closed, but looking at how the container binding works, I am of the opinion that the class bound by the container can change, and therefore not always be of a 'LogManager' instance. My patch would still let everything work but also prevent the bug from occurring...
Not much more I can do, I'll create a workaround in my codebase for it and hope this get patched or something in the future.
